data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java

```diff
@@ -144,8 +144,8 @@ public void createInputFile() throws IOException {
       record.setField("_boolean", false);
       record.setField("_int", i);
       record.setField("_long", 5_000_000_000L + i);
-      record.setField("_float", ((float) (100 - i)) / 100F + 1.0F); // 2.0f, 1.99f, 1.98f, ...
-      record.setField("_double", ((double) i) / 100.0D + 2.0D); // 2.0d, 2.01d, 2.02d, ...
+      record.setField("_float", (i % 10 == 0) ? Float.NaN : ((float) (100 - i)) / 100F + 1.0F); // 2.0f, 1.99f, 1.98f, ... OR NAN
+      record.setField("_double", (i % 10 == 0) ? Double.NaN : ((double) i) / 100.0D + 2.0D); // 2.0d, 2.01d, 2.02d, ... OR NAN
```

Comment:
We probably shouldn't do that here. we can do negative tests as long as there aren't NANs correct?

Reply:
In testEq() it checks that the first parameterized value is read, and the second value is not read. So if there are NANs this should fail, if it doesn't fail with this code change let's figure out why.